app/src/main/java/com/muzima/view/forms/HTMLFormDataStore.java

```diff
@@ -549,18 +551,38 @@ public Boolean isLocationServicesEnabled(){
     }
 
     public void showLocationDisabledDialog(){
-        AlertDialog.Builder alertDialog = new AlertDialog.Builder(application);
-        alertDialog.setTitle("Enable Location and Internet connectivity.");
-        alertDialog.setMessage("Location is switched off! Kindly turn location on in settings.");
-        alertDialog.setPositiveButton("Location Settings", new DialogInterface.OnClickListener() {
-            public void onClick(DialogInterface dialog, int which) {
-                Intent intent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
-                application.startActivity(intent);
-            }
-        });
+        Boolean isGPSDataCollectionEnabledSetting = new GPSFeaturePreferenceService(application).getIsGPSDataCollectionEnabledSetting();
```

Comment:
I think this could be renamed to isGPSDataCollectionSettingEnabled and isGPSDataCollectionSettingEnabled()